back-end/routes/api.js

```diff
@@ -1,13 +1,25 @@
-const express = require('express');
+const fs = require("fs");
+const path = require("path");
+const express = require("express");
 const router = express.Router();
 
 router.get("/test", (req, res) => {
   res.status(200).send({ message: "OK" });
 });
 
 router.post("/signin", (req, res) => {
-  if (req.body.username === "test" && req.body.password === "admin") {
-    res.status(200).send({ token: "pizzaboi" });
+  let accounts;
+  try {
+    accounts = JSON.parse(
+      fs.readFileSync(path.join(__dirname, "../../accounts.json"))
+    );
+  } catch (error) {
+    throw new Error(`Error while reading accounts from file: ${error}`);
+  }
+  const username = req.body.username;
```

Comment:
check whether username has a value after this, and if it doesnt, send a 400 response (bad request) with an appropriate message